tests/wpt/web-platform-tests/html/syntax/parsing/the-end.html

```diff
@@ -29,12 +29,28 @@
 }, "load");
 
 async_test(function() {
-  var seen = false;
+  window.addEventListener("pageshow", this.step_func_done(function(e) {
+    assert_equals(e.type, "pageshow");
+    assert_false(e.bubbles, "bubbles should be false");
+    assert_false(e.cancelable, "cancelable should be false");
+    assert_equals(e.target, document, "target should be document");
+    assert_true(e.isTrusted, "isTrusted should be true");
+    assert_class_string(e, "PageTransitionEvent");
+  }));
+}, "pageshow");
+
+async_test(function() {
+  var seen_dcl = false;
+  var seen_load = false;
   document.addEventListener("DOMContentLoaded", this.step_func(function() {
-    seen = true;
+    seen_dcl = true;
   }));
   window.addEventListener("load", this.step_func_done(function() {
```

Comment:
step_func_done here will cause the async_test to finish. This should use step_fun, and the pageshow event should use step_func_done instead.